src/main/java/org/folio/rest/impl/VendorHelper.java

```diff
@@ -0,0 +1,194 @@
+package org.folio.rest.impl;
+
+import io.vertx.core.Context;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import me.escoffier.vertx.completablefuture.VertxCompletableFuture;
+import org.folio.HttpStatus;
+import org.folio.orders.rest.exceptions.HttpException;
+import org.folio.orders.utils.ErrorCodes;
+import org.folio.rest.jaxrs.model.CompositePurchaseOrder;
+import org.folio.rest.jaxrs.model.Error;
+import org.folio.rest.jaxrs.model.Errors;
+import org.folio.rest.tools.client.interfaces.HttpClientInterface;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toList;
+import static org.folio.orders.utils.ErrorCodes.GENERIC_ERROR_CODE;
+import static org.folio.orders.utils.ErrorCodes.ORDER_VENDOR_IS_INACTIVE;
+import static org.folio.orders.utils.ErrorCodes.ORDER_VENDOR_NOT_FOUND;
+import static org.folio.orders.utils.ErrorCodes.POL_ACCESS_PROVIDER_IS_INACTIVE;
+import static org.folio.orders.utils.ErrorCodes.POL_ACCESS_PROVIDER_NOT_FOUND;
+import static org.folio.orders.utils.HelperUtils.convertIdsToCqlQuery;
+import static org.folio.orders.utils.HelperUtils.encodeQuery;
+import static org.folio.orders.utils.HelperUtils.handleGetRequest;
+
+
+public class VendorHelper {
+
+  private static final Logger logger = LoggerFactory.getLogger(VendorHelper.class);
+
+  private final HttpClientInterface httpClient;
+  private final Map<String, String> okapiHeaders;
+  private final Context ctx;
+  private final String lang;
+
+  private static final String ID = "id";
+  private static final String VENDORS = "vendors";
+  private static final String VENDOR_STATUS = "vendor_status";
```

Comment:
My personal preference is to use [mod-vendors schemas](https://github.com/folio-org/acq-models/tree/master/mod-vendors/schemas) instead of `JsonObject`. This requires extra conversion but it allows to catch potential issues at compile time rather than run-time.